src/QuadForm/Herm/Spaces.jl

```diff
@@ -202,17 +202,24 @@ end
 ################################################################################
 
 function isisometric(M::HermSpace, L::HermSpace)
+  @req isregular(M) && isregular(L) "The spaces must be both regular"
+  @req base_ring(M) === base_ring(L) "The spaces must be defined over the same ring"
   if gram_matrix(M) == gram_matrix(L)
     return true
   end
 
   if rank(M) != rank(L)
     return false
   end
-
+  
   E = base_ring(M)
-  # I could replace this with a islocal_norm at the ramified primes + primes
-  # dividing right hand side
+  K = base_field(E)
+  infp = real_places(K)
+  
+  if !all(v -> isisometric(M, L, v), infp)
```

Comment:
```suggestion
  if any(v -> !isisometric(M, L, v), infp)
```
This is a bit more efficient.